packages/lambdas/src/sqs-to-fhir.ts

```diff
@@ -208,6 +212,11 @@ export async function handler(event: SQSEvent) {
           "likely not to get completed - it might need manual intervention",
       },
     });
+
+    for (const record of event.Records) {
+      await changeMessageVisibility(record, 0);
+    }
```

Comment:
I was reading about this yesterday ([link](https://www.ranthebuilder.cloud/post/mastering-retries-best-practices-for-amazon-sqs)) and one important thing is to determine the type of error:
- recoverable: those are network issues or similar, things that temporary due to the nature of distributed systems;
- non-recoverable: e.g., if the downstream service returned 4xx b/c the shape of the payload is incorrect, there's no point in retrying.

Then, we can update the visibility timeout to 0 for non-recoverable errors and let SQS send the message to the DLQ.

But we prob want to let it retry the message if it's recoverable. The thing is, we need to have retries set, and might want to decrease the visibility timeout to maybe half so it doesn't take 30 min on a lambda that has 15min of max execution time.

Reply:
We already have `executeWithNetworkRetries` going in this code that is making this distinction so I think we are good here